src/tribler/core/utilities/tests/test_pony_utils.py

```diff
@@ -0,0 +1,88 @@
+import re
+from unittest.mock import patch
+
+import pytest
+
+from pony.orm.core import QueryStat, Required, commit, rollback
+from tribler.core.utilities import pony_utils
+
+
+EMPTY_DICT = {}
+
+
+def test_merge_stats_empty_iter():
+    stats_iter = []
+    merged_stats = pony_utils.TriblerDbSession._merge_stats(stats_iter)  # pylint: disable=protected-access
+    assert merged_stats == EMPTY_DICT  # to satisfy linter
+
+
+def test_merge_stats():
+    stats1 = {
+        None: QueryStat(None, duration=3.0),  # aggregated stats for database 1
+        "SQL1": QueryStat("SQL1", duration=1.0),
+        "SQL2": QueryStat("SQL2", duration=2.0),
+        "SQL3": QueryStat("SQL3", duration=3.0),
+    }
+    stats2 = {
+        None: QueryStat(None, duration=3.0),  # aggregated stats for database 2
+        "SQL2": QueryStat("SQL2", duration=3.0),
+        "SQL3": QueryStat("SQL3", duration=2.0),
+        "SQL4": QueryStat("SQL4", duration=4.0),
+    }
+    stats_iter = [stats1, stats2]
+    merged_stats = pony_utils.TriblerDbSession._merge_stats(stats_iter)  # pylint: disable=protected-access
+    max_times = {sql: stat.max_time for sql, stat in merged_stats.items()}
+    assert max_times == {
+        None: pytest.approx(3.0),
+        "SQL1": pytest.approx(1.0),
+        "SQL2": pytest.approx(3.0),
+        "SQL3": pytest.approx(3.0),
+        "SQL4": pytest.approx(4.0),
+    }
+
+
+def test_patched_db_session(caplog, tmp_path):
+    # The test is added for better coverage of TriblerDbSession methods
+
+    with patch('pony.orm.dbproviders.sqlite.provider_cls', pony_utils.PatchedSQLiteProvider):
+        db = pony_utils.TriblerDatabase()
+        db.bind('sqlite', str(tmp_path / 'db.sqlite'), create_db=True)
+
+        class Entity1(db.Entity):
+            a = Required(int)
+
+        db.generate_mapping(create_tables=True)
+
+        @pony_utils.db_session
+        def perform_queries():
+            for i in range(1000):
+                Entity1(a=i)
+            commit()
+            rollback()
+            Entity1.select().fetch()
+
+        caplog.clear()
+        with patch('tribler.core.utilities.pony_utils.SLOW_DB_SESSION_DURATION_THRESHOLD', 0.0):
```

Comment:
This line illustrates that the pony_utils are difficult to customize, as you resort to a patch to alter the SLOW_DB_SESSION_DURATION_THRESHOLD value.

I would recommend revising how the utils can be customized.

Reply:
I added the keyword argument `db_session(duration_threshold=...)` to not patch the constant here